pkg/assisted/nmstateconfig.go

```diff
@@ -0,0 +1,229 @@
+		Definition: &assistedv1beta1.NMStateConfig{
+			ObjectMeta: metaV1.ObjectMeta{
+				Name:      name,
+				Namespace: namespace,
+			},
+		},
+	}
+
+	if name == "" {
+		glog.V(100).Infof("The name of the nmstateconfig is empty")
+
+		builder.errorMsg = "nmstateconfig 'name' cannot be empty"
+	}
+
+	if namespace == "" {
+		glog.V(100).Infof("The namespace of the nmstateconfig is empty")
+
+		builder.errorMsg = "nmstateconfig namespace's name is empty"
+	}
+
+	return &builder
+}
+
+// Exists checks whether the given NMStateConfig exists.
+func (builder *NmStateConfigBuilder) Exists() bool {
+	if valid, _ := builder.validate(); !valid {
+		return false
+	}
+
+	glog.V(100).Infof("Checking if nmstateconfig %s exists in namespace: %s",
+		builder.Definition.Name, builder.Definition.Namespace)
+
+	var err error
+	builder.Object, err = builder.Get()
+
+	return err == nil || !k8serrors.IsNotFound(err)
+}
+
+// Get returns NMStateConfig object if found.
+func (builder *NmStateConfigBuilder) Get() (*assistedv1beta1.NMStateConfig, error) {
+	if valid, err := builder.validate(); !valid {
+		return nil, err
+	}
+
+	glog.V(100).Infof("Collecting nmstateconfig object %s in namespace: %s",
+		builder.Definition.Name, builder.Definition.Namespace)
+
+	nmStateConfig := &assistedv1beta1.NMStateConfig{}
+	err := builder.apiClient.Get(context.TODO(), goclient.ObjectKey{
+		Name:      builder.Definition.Name,
+		Namespace: builder.Definition.Namespace,
+	}, nmStateConfig)
+
+	if err != nil {
+		glog.V(100).Infof("nmstateconfig object %s doesn't exist", builder.Definition.Name)
+
+		return nil, err
+	}
+
+	return nmStateConfig, err
+}
+
+// Create makes a NMStateConfig in the cluster and stores the created object in struct.
+func (builder *NmStateConfigBuilder) Create() (*NmStateConfigBuilder, error) {
+	if valid, err := builder.validate(); !valid {
+		return builder, err
+	}
+
+	glog.V(100).Infof("Creating the nmstateconfig %s in namespace: %s",
+		builder.Definition.Name, builder.Definition.Namespace)
+
+	var err error
+	if !builder.Exists() {
+		err = builder.apiClient.Create(context.TODO(), builder.Definition)
+		if err == nil {
+			builder.Object = builder.Definition
+		}
+	}
+
+	return builder, err
+}
+
+// Delete removes nmstateconfig object from a cluster.
+func (builder *NmStateConfigBuilder) Delete() (*NmStateConfigBuilder, error) {
+	if valid, err := builder.validate(); !valid {
+		return builder, err
+	}
+
+	glog.V(100).Infof("Deleting the nmstateconfig object %s in namespace: %s",
+		builder.Definition.Name, builder.Definition.Namespace)
+
+	err := builder.apiClient.Delete(context.TODO(), builder.Definition)
+
+	if err != nil {
+		return builder, fmt.Errorf("can not delete nmstateconfig: %w", err)
+	}
+
+	builder.Object = nil
+
+	return builder, nil
+}
+
+// validate will check that the builder and builder definition are properly initialized before
+// accessing any member fields.
+func (builder *NmStateConfigBuilder) validate() (bool, error) {
```

Comment:
Please ensure that private functions and methods are listed at the bottom of the file